### ISSUE TITLE
Mismatch between lint definitions and Cloudformation for AWS::ECS::Service.NetworkConfiguration.AwsvpcConfiguration

### ISSUE THREAD

**Issue by p4k03n4t0r** [aws-cloudformation/cfn-python-lint#1706]
*cfn-lint version: (`cfn-lint --version`)*
0.36.0

*Description of issue.*
The linter gives an error, while Cloudformation disagrees. For the resource type AWS::ECS::Service there is a property NetworkConfiguration. According to the [ Cloudformation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-service-networkconfiguration.html) the property AwsvpcConfiguration has a lower case v (I confirmed this is actually the case by deploying a Cloudformation stack). The cfn-lint with version 0.36.0 expects  AwsVpcConfiguration with an upper case v. With earlier versions this didn't give any errors. Cloudformation will not accept an upper case v.

**Comment by PatMyron** [comment]
[Resource Specification](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html):
```json
    "AWS::ECS::Service.NetworkConfiguration": {
      "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-service-networkconfiguration.html",
      "Properties": {
        "AwsVpcConfiguration": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-service-networkconfiguration.html#cfn-ecs-service-networkconfiguration-awsvpcconfiguration",
          "UpdateType": "Mutable",
          "Required": false,
          "Type": "AwsVpcConfiguration"
        }
      }
    },
```
another similar ECS property we should double check as well:
```json
    "AWS::ECS::TaskSet.NetworkConfiguration": {
      "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskset-networkconfiguration.html",
      "Properties": {
        "AwsVpcConfiguration": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskset-networkconfiguration.html#cfn-ecs-taskset-networkconfiguration-awsvpcconfiguration",
          "UpdateType": "Immutable",
          "Required": false,
          "Type": "AwsVpcConfiguration"
        }
      }
    },
```

**Comment by PatMyron** [comment]
@p4k03n4t0r this is caused during the ongoing migration of [`AWS::ECS::Service`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-service.html) to the [CloudFormation Registry](https://aws.amazon.com/blogs/aws/cloudformation-update-cli-third-party-resource-support-registry/)

Can workaround with [resource-level ignores](https://github.com/aws-cloudformation/cfn-python-lint#resource-based-metadata):

```yaml
  Resource:
    Type: AWS::ECS::Service
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
          - E3002
```

**Comment by simongibbons** [comment]
I'm also affected by this and have used @PatMyron 's work around. I assume we need to wait for AWS to release a fixed version of the resource specifications?

**Comment by kddejong** [comment]
Or we patch the spec until this is fixed.

**Comment by joaquin386** [comment]
Or use 0.35.1 which is what we did.

**Comment by PatMyron** [comment]
> Or we patch the spec until this is fixed.

I'm not even sure which way we'd want to patch this:

Code generation tools built on the [Resource Specification](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) like [CDK](https://github.com/aws/aws-cdk/)/[Troposphere](https://github.com/cloudtools/troposphere/)/[GoFormation](https://github.com/awslabs/goformation/)/[IDE extensions](https://github.com/aws-cloudformation/aws-cfn-lint-visual-studio-code/)/Designer console/etc. will start generating templates with the new property casing while existing templates while have the old casing, and the old casing (lowercase `v`) is still required in regions that have not completed migration to the [CloudFormation Registry](https://aws.amazon.com/blogs/aws/cloudformation-update-cli-third-party-resource-support-registry/)

Property casing simply should not be changed during [CloudFormation Registry](https://aws.amazon.com/blogs/aws/cloudformation-update-cli-third-party-resource-support-registry/) migrations

**Comment by kddejong** [comment]
Almost sounds like we should patch in both options.  What happens in a region that has done the migration?  Does the old way no longer work?

**Comment by PatMyron** [comment]
> What happens in a region that has done the migration? Does the old way no longer work?

Been told [CloudFormation Registry](https://aws.amazon.com/blogs/aws/cloudformation-update-cli-third-party-resource-support-registry/) property types are case insensitive, unlike previous frameworks, so the old casing should still deploy
